tests/test_integration.py

```diff
@@ -378,3 +378,362 @@ def test(self, soco):
             'No difference between '
             'queues before and after removing the last item')
         assert len(new_queue) == len(old_queue) - 1
+
+
+class TestReorderSonosPlaylist(object):
+    """ Integration tests for Sonos Playlist Management. """
+    existing_playlists = None
+    pl_name = 'zSocoTestPlayList42'
+    test_pl = None
+
+    def _get_test_pl(self, soco):
+        '''return the DidlPlayListContainer for the pl_name'''
+        for spl in soco.get_sonos_playlists():
+            if spl.title == self.pl_name:
+                return spl
+
+    @pytest.yield_fixture(autouse=True, scope="class")
+    def restore_sonos_playlists(self, soco):
+        """ A fixture which cleans up after each sonos playlist test."""
+        if self.existing_playlists is None:
+            self.existing_playlists = soco.get_sonos_playlists()
+        if self.pl_name in [x.title for x in self.existing_playlists]:
+            msg = '%s is an existing playlist.' % self.pl_name
+            pytest.fail(msg)
+
+        queue_list = soco.get_queue()
+        if len(queue_list) < 2:
+            msg = 'You must have 3 or more items in your queue for testing.'
+            pytest.fail(msg)
+        self.test_pl = soco.create_sonos_playlist_from_queue(self.pl_name)
+        yield
+
+        # soco.remove_sonos_playlist(object_id=self.test_pl.item_id)
+        soco.contentDirectory.DestroyObject([('ObjectID',
+                                              self.test_pl.item_id)])
+
+    def _reset_spl_contents(self, soco):
+        """ Ensure test playlist matches queue for each test. """
+        update_id = 0
+        que = soco.get_queue()
+        test_pl = self._get_test_pl(soco)
+        spl = soco.browse(ml_item=test_pl)
+        tracks = list(reversed(range(len(spl))))
+        new_pos = [None for _ in tracks]
+        if tracks:
+            _, count, update_id = soco.reorder_sonos_playlist(
+                                            object_id=test_pl.item_id,
+                                            tracks=tracks,
+                                            new_pos=new_pos,
+                                            update_id=update_id)
+            if count:
+                msg = 'Failed to empty playlist. Unexpected result.'
+                pytest.fail(msg)
+        for track in que:
+            soco.add_item_to_sonos_playlist(track, test_pl)
+        return test_pl
+
+    def test_reverse_track_order(self, soco):
+        """ Test reversing the tracks in the Sonos playlist. """
```

Comment:
@dundeemt Yeeess :$ not particularly consistent up to this, sorry that this bites you.